src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/FileResultExecutorBase.cs

```diff
@@ -229,6 +238,11 @@ private static void SetAcceptRangeHeader(HttpResponse response)
             {
                 var unmodified = ifUnmodifiedSince >= lastModified;
                 ifUnmodifiedSinceState = unmodified ? PreconditionState.ShouldProcess : PreconditionState.PreconditionFailed;
+
+                if (ifUnmodifiedSinceState == PreconditionState.PreconditionFailed)
+                {
+                    logger.IfUnmodifiedSincePreconditionFailed(lastModified, ifUnmodifiedSince);
+                }
```

Comment:
There should also be logging for the case above where ifModifiedSinceState is NotModified

Reply:
I didn't want to add logging for this case as its usually pretty clear if a response is 304, which is not modified. I felt PreconditionFailed to be interesting as `GetMaxPreconditionState` tries to find a max of all states and if a precondition failed, a user might be interested to know why.